aion/web/channels/user_socket.ex

```diff
@@ -10,15 +11,18 @@ defmodule Aion.UserSocket do
 
   transport :websocket, Phoenix.Transports.WebSocket
 
-  def connect(%{"token" => token}, socket) do
-    case Token.verify(socket, "user", token, max_age: 86_400) do
-      {:ok, user_id} ->
-        socket = assign(socket, :current_user, Repo.get!(User, user_id))
+  def connect(%{"token" => jwt}, socket) do
+    case sign_in(socket, jwt) do
+      {:ok, auth_socket, guardian_params} ->
+        {:ok, auth_socket}
+      _ ->
```

Comment:
Probably need to think about better error handling but Im not sure if that would pay off. The only case is where we there is either no token provided or it expired. For now I would leave it since frontend also does not have a solution for refreshing expired token.

I would make an issue of that - to improve the way we handle errors related to authentication. For now I will just return `:error` as suggested in docs: https://hexdocs.pm/phoenix/Phoenix.Socket.html#callbacks

Reply:
Yes, that's exactly what I was talking about - the callback saying there may be a case for throwing an error. You should probably make it return error in those cases then and also create the said issue.